acra/configuring-maintaining/optimisations/acrastructs_vs_acrablocks.md

```diff
@@ -0,0 +1,27 @@
+---
+title: AcraStructs vs AcraBlocks
+weight: 3
+---
+
+# AcraStructs vs AcraBlocks
+
+There are two different "crypto containers" AcraServer can use to store encrypted value in.
+Then, these containers are stored in database and decrypted back by AcraServer when needed.
+
+History
+* `AcraStruct` is our "old" one, though we're not going to remove it any time soon.
+* `AcraBlock` on the other hand is new and proposes some extra features.
+
+Encryption with a key
+* `AcraStruct` uses key provided from somewhere outside (by Acra) to encrypt the value.
```

Comment:
uses storage public key that related to private key known only for AcraServer. Main difference between AcraStruct and AcraBlock that first uses asymmetric key encryption that are slow due to multiplications of big numbers (elliptic curve encryption) and symmetric key encryption that just XOR arrays using optimized instructions in CPU.